src/UiPath.Workflow.Runtime/Statements/Delay.cs

```diff
@@ -62,6 +71,9 @@ protected override void Cancel(NativeActivityContext context)
         timerExtension.CancelTimer(timerBookmark);
         context.RemoveBookmark(timerBookmark);
         context.MarkCanceled();
+
+        if (HasBlockingDelay(context))
```

Comment:
we shouldn't need it here either; see the NoPersistScope statement (activity); it doesn't bother calling Exit at all.